lnwallet/channel.go

```diff
@@ -552,6 +552,20 @@ func (lc *LightningChannel) diskHtlcToPayDesc(feeRate chainfee.SatPerKWeight,
 		remoteOutputIndex = htlc.OutputIndex
 	}
 
+	customRecords := htlc.CustomRecords.Copy()
+
+	entryType := Add
+
+	noopTLV := uint64(noopHtlcType.TypeVal())
+	if _, ok := customRecords[noopTLV]; ok {
+		entryType = NoopAdd
+	}
+
+	// The NoopAdd HTLC is an internal construct, and isn't meant to show up
+	// on the wire. So we'll remove the special element from the set of
+	// custom records.
+	delete(customRecords, noopTLV)
```

Comment:
IIUC this wouldn't really matter, as the receiver of the wire message would just ignore this unknown type, are we just saving wire space here?

Reply:
Ended up doing this as we have some tests that assert what an HTLC should look like for transmission. If we continue to thread these over, then those start to fail. 

On the other hand, maybe it makes sense to expose this, as then the receiver just accepts the noop if the sender does, vs the current logic of assuming that it's always noop when the tapscript bit is set.